### PR TITLE
[15.0][IMP] hr_attendance_reason: Show reasons on attendance screen

### PR DESCRIPTION
**Pull request by victoralmau**
FWP from 14.0: https://github.com/OCA/hr-attendance/pull/102 + https://github.com/OCA/hr-attendance/pull/123 + https://github.com/OCA/hr-attendance/pull/125

Show reasons on attendance screen.
ejemplo

@Tecnativa TT40897

### CODE REVIEW

**Comment by pedrobaeza** [comment]
You can squash 3 first commits IMO.

**Comment by victoralmau** [comment]
Now, you are ready to review. My tests everything works correctly.

**Comment by victoralmau** [comment]
Ping @chienandalu

File: hr_attendance_reason/static/src/js/kiosk_confirm.esm.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        await this._rpc({
+            model: "hr.attendance.reason",
+            method: "search_read",
+            fields: ["name", "action_type"],
+            domain: [["show_on_attendance_screen", "=", true]],
+        }).then((reasons) => {
+            this.reasons = reasons;
+        });
+    },
+    _rpc: function () {
```
Comment (chienandalu) [inline hr_attendance_reason/static/src/js/kiosk_confirm.esm.js:44]: Let's comment this override as the way to inject into the context of an enclosed rpc call

File: hr_attendance_reason/static/src/js/my_attendances.esm.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            model: "hr.attendance.reason",
+            method: "search_read",
+            fields: ["name", "action_type"],
+            domain: [["show_on_attendance_screen", "=", true]],
+        }).then((reasons) => {
+            this.reasons = reasons;
+        });
+        return this.reasons;
+    },
+    _rpc: function () {
```
Comment (chienandalu) [inline hr_attendance_reason/static/src/js/my_attendances.esm.js:33]: Same here

**Review by chienandalu — APPROVED** [review]

File: hr_attendance_reason/models/hr_employee_base.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2023 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class HrEmployeeBase(models.AbstractModel):
+    _inherit = "hr.employee.base"
+
+    show_reasons_on_attendance_screen = fields.Boolean(
```
Comment (pedrobaeza) [inline hr_attendance_reason/models/hr_employee_base.py]: Please avoid plurals in variable names.

File: hr_attendance_reason/models/res_company.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Copyright 2023 Tecnativa - Víctor Martínez
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import fields, models
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    show_reasons_on_attendance_screen = fields.Boolean(
```
Comment (pedrobaeza) [inline hr_attendance_reason/models/res_company.py]: Avoid plurals.

**Comment by victoralmau** [comment]
Changes done.

**Pull request merged**